MiniRazor.Compiler/Razor.cs

```diff
@@ -111,6 +111,18 @@ void PopulateTransitiveDependencies(Assembly assembly, ISet<AssemblyName> assemb
                 .TryLoadFromAssemblyName(typeof(TemplateBase<>).Assembly.GetName())?
                 .ToMetadataReference();
 
+            if (razorCompileOptions?.InferReferencesFromAssemblyLoadContext == true)
+            {
+#if NETCOREAPP3_0_OR_GREATER
```

Comment:
At the moment, if this is used from the other TFMs nothing happens, but I see this being relatively less likely, and this is something that users will knowingly opt-in to.

I felt that having conditional compilation at the options class, or API overloads would be an uglier option.